src/tracer/MARBL_tracers.F90

```diff
@@ -0,0 +1,734 @@
+! their mks counterparts with notation like "a velocity [Z T-1 ~> m s-1]".  If the units
+! vary with the Boussinesq approximation, the Boussinesq variant is given first.
+
+!> Temporary type for diagnostic variables coming from MARBL
+!! Allocate exactly one of field_[23]d
+type, private :: temp_MARBL_diag
+  integer :: id !< index into MOM diagnostic structure
+  real, allocatable :: field_2d(:,:) !< memory for 2D field
+  real, allocatable :: field_3d(:,:,:) !< memory for 3D field
+end type temp_MARBL_diag
+
+type, private :: saved_state_for_MARBL_type
+  character(len=200) :: short_name !< name of variable being saved
+  character(len=200) :: file_varname !< name of variable in restart file
+  character(len=200) :: units !< variable units
+  real, allocatable :: field_2d(:,:) !< memory for 2D field
+  real, allocatable :: field_3d(:,:,:) !< memory for 3D field
+end type saved_state_for_MARBL_type
+
+!> All calls to MARBL are done via the interface class
+type(MARBL_interface_class), private :: MARBL_instances
+
+!> The control structure for the MARBL tracer package
+type, public :: MARBL_tracers_CS ; private
+  integer :: ntr    !< The number of tracers that are actually used.
+  logical :: coupled_tracers = .false.  !< These tracers are not offered to the coupler.
+  character(len=200) :: IC_file !< The file in which the age-tracer initial values cam be found.
+  type(tracer_registry_type), pointer :: tr_Reg => NULL() !< A pointer to the tracer registry
+  real, pointer :: tr(:,:,:,:) => NULL() !< The array of tracers used in this subroutine, in g m-3?
+
+  integer, allocatable, dimension(:) :: ind_tr !< Indices returned by aof_set_coupler_flux if it is used and the
+                                               !! surface tracer concentrations are to be provided to the coupler.
+
+  type(diag_ctrl), pointer :: diag => NULL() !< A structure that is used to
+                                   !! regulate the timing of diagnostic output.
+  type(MOM_restart_CS), pointer :: restart_CSp => NULL() !< A pointer to the restart control structure
+
+  type(vardesc), allocatable :: tr_desc(:) !< Descriptions and metadata for the tracers
+  logical :: tracers_may_reinit = .false. !< If true the tracers may be initialized if not found in a restart file
+
+  !> Driver-specific parameters
+  character(len=35) :: marbl_settings_file
+  real :: atm_co2_const, atm_alt_co2_const
+
+  !> Indices to the registered diagnostics match the indices used in MARBL
+  type(temp_MARBL_diag), allocatable :: surface_flux_diags(:) !, interior_tendency_diags(:)
+  type(saved_state_for_MARBL_type), allocatable :: surface_flux_saved_state(:) !, interior_tendency_saved_state(:)
+  integer :: u10_sqr_ind, sss_ind, sst_ind, ifrac_ind, dust_dep_ind, fe_dep_ind
+  integer :: nox_flux_ind, nhy_flux_ind, atmpress_ind, xco2_ind, xco2_alt_ind
+end type MARBL_tracers_CS
+
+!> If we can post data column by column, all we need are integer
```

Comment:
I don't think you can post it column by column, what you could do is make a 3D array and then when you feed in the diagnostic array for the column, just pass it by pointer.
```
real, dimension(:,:,:), allocatable :: diag_3d
real, dimension(:), pointer :: diag_3d_ptr

do i=isc;iec; do j=jsc;jec
  diag_3d_ptr => diag_3d(i,j,:)
  call MARBL_(...,diag_3d_ptr)

enddo
```

Reply:
> I understand what you're trying to do here and I largely agree with the implementation. There are some minor stylistic kind of things suggested in this review. For the `FIXME` and `TODO` do you want me to offer suggestions on how to implement with some of MOM6 capabilities?

@ashao Thanks for all these comments! I've replied to most of them; the ones I did not reply just didn't fit into either the "I know the answer to this question" category or the "oh, that's an easy change to make" category. I'll resolve the 👍  comments after committing the fixes.

I thought I answered the `FIXME` / `TODO` question, but I'm not seeing it. So apologies if I'm repeating myself (and double apologies if this contradicts the previous response that I can't find) but suggestions on implementing fixes would be great!